### DISCUSSION TITLE
What creates the daily ElasticSearch Indexes and mappings

### DISCUSSION THREAD

**Discussion by MarkBaggett** [Security-Onion-Solutions/securityonion#2538] [Unsupported Versions] [answered]
I've created an elasticsearch custom pipeline to ingest domainstats logs generated by zeek.  Before the data appears in elasticsearch I had to create a mapping (shown below).   After doing this it will work for 1 day.

PUT /so-zeek-2021-01-04/_mapping
{
  "properties", {
      "domainstats": {
             "type": "object",
              "dynamic", "true"
        }
   }
}

This was done via the developers tools in elastic. 
What is the proper way to add this so that it is added each day?  Does adding my pipeline file to /opt/so/conf/eleasticsearch/ingest do it?   

Thanks,
Mark

**Comment by weslambert** [comment, marked as answer]
Hi Mark,

You'll want to create a new template in `/opt/so/saltstack/local/salt/elasticsearch/templates/custom` (have a look in `/opt/so/saltstack/default/salt/elasticsearch/templates/so` for examples), copy `/opt/so/saltstack/default/pillar/elasticsearch/search.sls` to `/opt/so/saltstack/local/pillar/elasticsearch/search.sls`, and restart Elasticsearch.

After that, you should be able to list the loaded templates with `so-elasticsearch-templates-list`

To be clear, the ingest pipeline files will allow the incoming data to be processed, but you will need the template to enforce the mapping.  Alternatively, you could copy the `so-common` template to the `local` directory referenced above, and simply add your single field mapping.

**Reply by MarkBaggett** [reply]
Thank you.   I'll try that.

**Reply by branden999** [reply]
Hey Wes,

I don't suppose there is more of a write-up on how to work with the ES templates somewhere is there? I found the one in the manual, but it hasn't helped much and it just doesn't seem to want to cooperate with me. I managed to get the templates to work on my older 16.x server, but since upgrading to 2.3.21 yesterday am having problems getting my head around it. I ran into a similar issue as the poster above which lead me to this thread and at least gave me a place to start.

Thanks,

Branden
